src/index.ts

```diff
@@ -3,7 +3,14 @@ import type { DataLayerObject } from './data-layer-object';
 declare global {
   interface Window {
     /**
-     * `dataLayer` used by GTM.
+     * Dynamic `dataLayer` key used by GTM.
+     *
+     * @see [developers.google.com/tag-manager/devguide/datalayer#rename_the_data_layer](https://developers.google.com/tag-platform/tag-manager/datalayer#rename_the_data_layer)
+     */
+    [key: string]: any;
```

Comment:
This is a really bad idea. Instead we should document how to add TS support for a custom key if really necessary/wanted.